src/c++/perf_analyzer/schema.json

```diff
@@ -0,0 +1,96 @@
+{
+    "$schema": "https://json-schema.org/draft/2020-12/schema",
+    "$id": "https://github.com/triton-inference-server/client/blob/main/src/c%2B%2B/perf_analyzer/examples/schema.json",
+    "title": "Perf Analyzer output data",
+    "description": "A json file describing the output from a perf analyzer run.",
+    "type": "object",
+    "required": [
+        "experiments",
+        "version"
+    ],
+    "properties": {
+        "experiments": {
+            "description": "The array of all experiments run by perf analyzer.",
+            "type": "array",
+            "required": [
+                "experiment",
+                "requests",
+                "window_boundaries"
+            ],
+            "minItems": 1,
+            "uniqueItems": true,
+            "items": {
+                "type": "object",
+                "properties": {
+                    "experiment": {
+                        "description": "A single experiment run by perf analyzer.",
+                        "type": "object",
+                        "required": [
+                            "mode",
+                            "value"
+                        ],
+                        "minItems": 1,
+                        "maxItems": 1,
+                        "properties": {
+                            "mode": {
+                                "description": "Operating mode of perf analyzer: For example, concurrency or request rate.",
+                                "type": "string"
+                            },
+                            "value": {
+                                "description": "Concurrency or request rate for the current experiment.",
+                                "type": "number"
+                            }
+                        }
+                    },
+                    "requests": {
+                        "description": "The array of requests sent by perf_analyzer for this experiment.",
+                        "type": "array",
+                        "uniqueItems": true,
+                        "items": {
```

Comment:
How come this uses `items` and `$ref` to define things, but above it would just be the name of the field in the `properties` key?

Reply:
Because of the way we defined the examples. The entries in the requests array are objects but they do not have key entries. If the requests entry was an object with key, value pairs, i could define this in the same manner as the rest. However, since the values are effectively anonymous, I need to define the way the objects are defined with a $ref. You can see here for an example: https://json-schema.org/learn/miscellaneous-examples.html#arrays-of-things

Reply:
Ooh, that makes sense. Thanks for the clarification :)

Reply:
You are very welcome. It took a bit to understand how to proper construct this schema to support it but we got there.